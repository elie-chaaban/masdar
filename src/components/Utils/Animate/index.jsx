import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import config from './config';

const Animate = ({enter, animation, unMountOnExit, options, children, onAnimationEnd, stylingProperties, ...props}) => {
  const [mount, setMount] = useState(false);
  const style = {
    animationName: `${animation}${enter ? 'In' : 'Out'}`,
    animationDuration: `${options.duration}ms`,
    animationDelay: `${options.delay}ms`,
    animationFillMode: options.fillMode,
    animationTimingFunction: options.timingFunction,
    animationPlayState: options.playState,
    animationDirection: options.direction,
    animationIterationCount: options.count
  };
  useEffect(() => {
    if (enter) setMount(true);
  }, [enter]);
  return mount || !unMountOnExit ? (
    <div
      {...props}
      style={{...style, ...stylingProperties}}
      onAnimationEnd={() => {
        if (!enter && unMountOnExit) {
          if (onAnimationEnd) onAnimationEnd();
          setMount(false);
        }
      }}
    >
      {children}
    </div>
  ) : null;
};

Animate.protoTypes = {
  enter: PropTypes.bool,
  animation: PropTypes.string,
  onAnimationEnd: PropTypes.func,
  unMountOnExit: PropTypes.bool,
  options: PropTypes.shape({
    duration: PropTypes.number,
    delay: PropTypes.number,
    fillMode: PropTypes.oneOf(config.fillMode),
    timingFunction: PropTypes.string,
    playState: PropTypes.oneOf(config.playState),
    direction: PropTypes.oneOf(config.direction),
    count: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(config.count)])
  })
};
Animate.defaultProps = {
  enter: true,
  animation: 'fade',
  unMountOnExit: false,
  onAnimationEnd: null,
  options: {
    duration: 1000,
    delay: 0,
    fillMode: 'forwards',
    timingFunction: 'linear',
    playState: 'running	',
    direction: 'normal',
    count: 'initial'
  }
};
export default Animate;
