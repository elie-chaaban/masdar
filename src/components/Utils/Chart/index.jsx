import React from 'react';
import Highcharts from 'highcharts';
import HighchartsMore from 'highcharts/highcharts-more';
import VariablePie from 'highcharts/modules/variable-pie';
import HighchartsReact from 'highcharts-react-official';
import useOptions from './options';

HighchartsMore(Highcharts);
VariablePie(Highcharts);

const Chart = (props) => {
  const options = useOptions(props);
  return (
    <div style={{width: props.width || 'auto'}} className="chart">
      <HighchartsReact highcharts={Highcharts} options={options} callback={props.onCreate} />
    </div>
  );
};

export default React.memo(Chart);
