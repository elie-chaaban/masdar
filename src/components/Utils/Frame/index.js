import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../../assets/images/logo/logo.svg';
import './styles.scss';

const Frame = ({source, showLogoDiv}) => {
  return (
    <>
      {showLogoDiv && (
        <div className="nxn-logo-div">
          <img className="nxn-logo-image" src={logo} alt="logo"></img>
        </div>
      )}
      <iframe src={source} frameBorder={'0'} style={{width: '100%', height: '100vh'}} title={'iframe'} />
    </>
  );
};

Frame.propTypes = {
  source: PropTypes.string.isRequired
};
export default Frame;
