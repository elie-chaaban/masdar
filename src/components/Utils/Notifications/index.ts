import {store, ReactNotificationOptions} from 'react-notifications-component';

export interface ErrorNotification {
  response: {
    status: number;
    statusText: string;
  };
}

export const successNotification = (message: string, type: ReactNotificationOptions['type'] = 'success') => {
  store.addNotification({
    title: type === 'default' ? 'Welcome' : type,
    message: message,
    type,
    insert: 'top',
    container: 'top-right',
    animationIn: ['fadeInRight'],
    animationOut: ['fadeOutRight'],
    dismiss: {
      duration: 3000,
      onScreen: true,
      showIcon: true
    }
  });
};

export const errorNotification = (error: ErrorNotification) => {
  const {response} = error;
  if (error.response) {
    if (response.status !== 401 && response.statusText) {
      if (response.status === 500) {
        store.addNotification({
          title: `Server Error`,
          message: 'Something went Wrong!',
          type: 'danger',
          insert: 'top',
          container: 'top-right',
          animationIn: ['animated', 'fadeInRight'],
          animationOut: ['animated', 'fadeOutRight'],
          dismiss: {
            duration: 5000,
            onScreen: true,
            showIcon: true
          }
        });
      } else {
        store.addNotification({
          title: `${response.status} Error `,
          message: response.statusText,
          type: 'danger',
          insert: 'top',
          container: 'top-right',
          animationIn: ['animated', 'fadeInRight'],
          animationOut: ['animated', 'fadeOutRight'],
          dismiss: {
            duration: 5000,
            onScreen: true,
            showIcon: true
          }
        });
      }
    }
  }
};
