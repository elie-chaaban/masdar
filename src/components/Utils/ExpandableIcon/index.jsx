import React from 'react';
import Icon from '../Icon';
import PropTypes from 'prop-types';
import Animate from '../Animate';

const ExpandableIcon = ({onClick, active, inverted = false, ...props}) => {
  return (
    <Animate enter={active} options={{duration: 500}} {...props} unMountOnExit>
      <Icon
        name={inverted ? 'ExpandableInverted' : 'Expandable'}
        className="expandable-icon clickable"
        onClick={onClick}
        width="25"
        height="50"
      />
    </Animate>
  );
};
ExpandableIcon.propTypes = {
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
  inverted: PropTypes.bool
};
export default React.memo(ExpandableIcon);
