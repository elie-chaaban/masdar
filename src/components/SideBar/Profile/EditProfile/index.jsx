import React, {useState, useEffect, useCallback} from 'react';
import {Formik, Form, Field} from 'formik';
import CountrySelect from '../CountrySelect';
import CitySelect from '../CitySelect';
import CustomInput from '../CustomInput';
import CountryCodeSelect from '../PhoneSelect';
import userEditSchema from '../validations/userEdit';
import {updateProfileInfo} from '../../../../services/user';
import {errorNotification, successNotification} from '../../../Utils/Notifications';

const EditProfile = ({countries, profile}) => {
  const {firstName, mobileNumber, city, address1 = '', address2 = '', email, gender} = profile;

  const cityId = city ? city?.id : null;
  const countryId = city ? city?.country?.id : null;
  const countryCode = city ? city?.country?.code : null;

  const [edit, setEdit] = useState(false);
  const [loading, setLoading] = useState(false);
  const [countryCities, setCountryCities] = useState([]);
  useEffect(() => {
    if (countries && city && city.id) {
      const selectedCountry = countries.find((c) => c.id === city.country.id);
      setCountryCities(selectedCountry.cities);
    }
  }, [countries, city]);
  const onChangeCountry = useCallback(
    (countryId) => {
      const selectedCountry = countries.find((c) => c.id === countryId);
      if (selectedCountry) {
        setCountryCities(selectedCountry.cities);
      } else {
        setCountryCities([]);
      }
    },
    [countries]
  );
  const updateProfile = async (gender, mobileNumberCountryCode, mobileNumber, cityId, address1, address2) => {
    try {
      setLoading(true);
      await updateProfileInfo(profile, gender, mobileNumberCountryCode, mobileNumber, cityId, address1, address2);
      setEdit(false);
      successNotification('Profile Updated Successfully!');
    } catch (e) {
      errorNotification(e);
    } finally {
      setLoading(false);
    }
  };
  return (
    <Formik
      initialValues={{
        userName: firstName,
        email: email,
        phone: mobileNumber,
        phoneCode: countryCode,
        country: countryId,
        city: cityId,
        address1: address1,
        address2: address2
      }}
      validationSchema={userEditSchema}
      onSubmit={({phone, phoneCode, country, city, address1, address2}) => {
        if (!loading) updateProfile(gender, phoneCode, phone, city, address1, address2);
      }}
    >
      {({errors, touched, resetForm}) => (
        <Form className="content row">
          <CustomInput name="userName" enabled={false} />
          <CustomInput name="email" type="email" enabled={false} />
          <div className="mt-2 col-12">
            <p>Phone</p>
          </div>
          <div className="col-6">
            <div className="row">
              <CountryCodeSelect name="phoneCode" enabled={edit} />
              <div className="col-9 pl-1">
                <Field
                  className={`form-control ${edit ? 'active' : ''}`}
                  type="number"
                  placeholder="phone number"
                  name="phone"
                  disabled={!edit}
                />
              </div>
              <div className="col-12">
                {((errors.phone && touched.phone) || (errors.phoneCode && touched.phoneCode)) && (
                  <div className="text-danger">{errors.phone || errors.phoneCode}</div>
                )}
              </div>
            </div>
          </div>
          <div className="col-6" />
          <CountrySelect
            countries={countries}
            error={errors.country}
            touched={touched.country}
            name="country"
            enabled={edit}
            onChange={onChangeCountry}
          />
          <CitySelect cities={countryCities} error={errors.city} touched={touched.city} name="city" enabled={edit} />
          <CustomInput name="address1" error={errors.address1} touched={touched.address1} title="address 1" enabled={edit} />
          <CustomInput name="address2" title="address 2" enabled={edit} />
          <div className="col-12 mt-3">
            {edit && (
              <button className="btn mr-2" type="submit">
                {loading ? 'Saving...' : 'Save'}
              </button>
            )}
            <button
              className="btn"
              type="button"
              onClick={() => {
                if (edit) resetForm();
                setEdit(!edit);
              }}
            >
              {edit ? 'Cancel' : 'Edit'}
            </button>
          </div>
        </Form>
      )}
    </Formik>
  );
};

export default React.memo(EditProfile);
