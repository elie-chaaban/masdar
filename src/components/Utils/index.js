import Animate from './Animate';
import Chart from './Chart';
import VideoStream from './VideoStream';
import SlideIn from './SlideIn';
import LoaderFull from './LoaderFull';
import Loader from './Loader';
import IndexWithIcon from './IndexWithIcon';
import Icon from './Icon';
import Frame from './Frame';
import ExpandableIcon from './ExpandableIcon';
import WhiteCloseIcon from './WhiteCloseIcon';
import GrayCloseIcon from './GrayCloseIcon';
import Legend from './Legend';
import {errorNotification, successNotification} from './Notifications';
import Dropdown from './MultilevelDropdown';

export {
  Frame,
  Animate,
  Chart,
  VideoStream,
  SlideIn,
  errorNotification,
  successNotification,
  LoaderFull,
  Loader,
  IndexWithIcon,
  Icon,
  ExpandableIcon,
  Dropdown,
  WhiteCloseIcon,
  GrayCloseIcon,
  Legend
};
