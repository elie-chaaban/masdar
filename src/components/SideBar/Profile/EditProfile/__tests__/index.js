import React from 'react';
import {getCountryCallingCode} from 'react-phone-number-input/input';
import {updateProfileInfo} from '../../../../../services/user';
import EditProfile from '..';
import {act} from 'react-dom/test-utils';
import {mount} from 'enzyme';
import {errorNotification, successNotification} from '../../../../Utils/Notifications';

jest.mock('../../../../Utils/Notifications', () => ({
  errorNotification: jest.fn(),
  successNotification: jest.fn()
}));

let resolve, reject;
jest.mock('react-phone-number-input/input', () => ({
  ...jest.requireActual('react-phone-number-input/input'),
  getCountries: () => ['LB', 'AE']
}));
jest.mock('../../../../../services/user', () => ({
  ...jest.mock('../../../../../services/user'),
  updateProfileInfo: jest.fn(
    () =>
      new Promise((res, rej) => {
        resolve = res;
        reject = rej;
      })
  )
}));
const profile = {
  firstName: 'adel96',
  mobileNumber: 9352,
  mobileNumberCountryCode: '961',
  address1: 'edde',
  gender: 'male',
  address2: '',
  email: 'adel@gmail.com',
  city: {id: 'cityId1', name: 'Dubai', country: {id: 'countryId1', code: '961', cities: [{id: 'cityId1', name: 'Beirut'}]}}
};
const countries = [
  {id: 'countryId1', name: 'Lebanon', code: '961', cities: [{id: 'cityId1', name: 'Beirut'}]},
  {id: 'countryId2', name: 'United Arab Emirates', code: '971', cities: [{id: 'cityId2', name: 'Dubai'}]}
];

describe('Edit Profile Component', () => {
  let dom;
  beforeEach(async () => {
    await act(async () => {
      dom = mount(<EditProfile profile={profile} countries={countries} />);
    });
  });
  afterEach(() => {
    dom.unmount();
    jest.clearAllMocks();
  });
  it('should match snapshot', () => {
    expect(dom).toMatchSnapshot();
  });
  it('should all input disabled by default and have passed props value', () => {
    expect(dom.find('input.active')).toHaveLength(0);
    expect(dom.find('select.active')).toHaveLength(0);
    expect(dom.find('button')).toHaveLength(1);
    expect(dom.find('button').text()).toBe('Edit');
    expect(dom.find('.text-danger')).toHaveLength(0);
    expect(dom.find('input[name="userName"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="email"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="phoneCode"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="phone"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="country"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="city"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="address1"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="address2"]').prop('disabled')).toBeTruthy();

    expect(dom.find('input[name="userName"]').prop('value')).toBe('adel96');
    expect(dom.find('input[name="email"]').prop('value')).toBe('adel@gmail.com');
    expect(dom.find('select[name="phoneCode"]').prop('value')).toBe('961');
    expect(dom.find('input[name="phone"]').prop('value')).toBe(9352);
    expect(dom.find('select[name="country"]').prop('value')).toBe('countryId1');
    expect(dom.find('select[name="city"]').prop('value')).toBe('cityId1');
    expect(dom.find('input[name="address1"]').prop('value')).toBe('edde');
    expect(dom.find('input[name="address2"]').prop('value')).toBe('');
  });
  test('Phone code select input', () => {
    const options = dom.find('select[name="phoneCode"]').find('option');
    expect(options).toHaveLength(3);
    expect(options.at(0).prop('value')).toBe('');
    expect(options.at(1).text()).toBe('' + getCountryCallingCode('LB'));
    expect(options.at(0).text()).toBe('code');
    expect(options.at(1).prop('value')).toBe(getCountryCallingCode('LB'));
    expect(options.at(2).text()).toBe('' + getCountryCallingCode('AE'));
    expect(options.at(2).prop('value')).toBe(getCountryCallingCode('AE'));
  });
  test('Country select input', () => {
    const options = dom.find('select[name="country"]').find('option');
    expect(options).toHaveLength(3);
    expect(options.at(0).text()).toBe('select country');
    expect(options.at(0).prop('value')).toBe('');
    expect(options.at(1).text()).toBe('Lebanon');
    expect(options.at(1).prop('value')).toBe('countryId1');
    expect(options.at(2).text()).toBe('United Arab Emirates');
    expect(options.at(2).prop('value')).toBe('countryId2');
  });
  test('City select input', () => {
    const options = dom.find('select[name="city"]').find('option');
    expect(options).toHaveLength(1);
    expect(options.at(0).text()).toBe('select city');
    expect(options.at(0).prop('value')).toBe('');
  });
  it('should enable all inputs on edit button click', async () => {
    await act(async () => {
      dom.find('button').simulate('click');
    });
    dom.update();
    expect(dom.find('button')).toHaveLength(2);
    expect(dom.find('button').at(0).text()).toBe('Save');
    expect(dom.find('button').at(1).text()).toBe('Cancel');
    expect(dom.find('input.active')).toHaveLength(3);
    expect(dom.find('select.active')).toHaveLength(3);
    expect(dom.find('input[name="userName"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="email"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="phoneCode"]').prop('disabled')).toBeFalsy();
    expect(dom.find('input[name="phone"]').prop('disabled')).toBeFalsy();
    expect(dom.find('select[name="country"]').prop('disabled')).toBeFalsy();
    expect(dom.find('select[name="city"]').prop('disabled')).toBeFalsy();
    expect(dom.find('input[name="address1"]').prop('disabled')).toBeFalsy();
    expect(dom.find('input[name="address2"]').prop('disabled')).toBeFalsy();
  });
  it('should validate required input fields', async () => {
    await act(async () => {
      await dom.find('button').simulate('click');
      await dom.find('select[name="country"]').simulate('change', {target: {value: '', name: 'country'}});
      await dom.find('input[name="phone"]').simulate('change', {target: {value: '', name: 'phone'}});
      await dom.find('select[name="city"]').simulate('change', {target: {value: '', name: 'city'}});
      await dom.find('input[name="address1"]').simulate('change', {target: {value: '', name: 'address1'}});
      dom.find('button').at(0).simulate('submit');
    });
    dom.update();
    expect(dom.find('select[name="country"]').prop('value')).toBe('');
    expect(dom.find('.text-danger')).toHaveLength(4);
    dom.find('.text-danger').forEach((text) => {
      expect(text.text()).toBe('required');
    });
    await act(async () => {
      await dom.find('input[name="phone"]').simulate('change', {target: {value: 123, name: 'phone'}});
      await dom.find('select[name="phoneCode"]').simulate('change', {target: {value: '', name: 'phoneCode'}});
      dom.find('button').at(0).simulate('submit');
    });
    dom.update();
    expect(dom.find('.text-danger')).toHaveLength(4);
  });
  test('form submit', async () => {
    await act(async () => {
      await dom.find('button').simulate('click');
      await dom.find('select[name="country"]').simulate('change', {target: {value: 'countryId1', name: 'country'}});
      await dom.find('input[name="phone"]').simulate('change', {target: {value: 7629, name: 'phone'}});
      await dom.find('select[name="city"]').simulate('change', {target: {value: 'cityId1', name: 'city'}});
      await dom.find('input[name="address1"]').simulate('change', {target: {value: 'port', name: 'address1'}});
      dom.find('button').at(0).simulate('submit');
    });
    dom.update();
    expect(dom.find('button').at(0).text()).toBe('Saving...');
    await act(async () => {
      resolve();
    });
    dom.update();
    expect(dom.find('button')).toHaveLength(1);
    expect(dom.find('button').text()).toBe('Edit');
    expect(successNotification).toBeCalledTimes(1);
    expect(successNotification).toBeCalledWith('Profile Updated Successfully!');
    expect(errorNotification).toBeCalledTimes(0);
    expect(updateProfileInfo).toBeCalledTimes(1);
    expect(updateProfileInfo).toBeCalledWith(expect.any(Object), 'male', '961', 7629, 'cityId1', 'port', '');
  });
  test('error on form submit ', async () => {
    await act(async () => {
      await dom.find('button').simulate('click');
      await dom.find('select[name="country"]').simulate('change', {target: {value: 'countryId1', name: 'country'}});
      await dom.find('input[name="phone"]').simulate('change', {target: {value: 7629, name: 'phone'}});
      await dom.find('select[name="city"]').simulate('change', {target: {value: 'cityId1', name: 'city'}});
      await dom.find('input[name="address1"]').simulate('change', {target: {value: 'port', name: 'address1'}});
      dom.find('button').at(0).simulate('submit');
    });
    await act(async () => {
      reject({response: {status: 500}});
    });
    expect(errorNotification).toBeCalledTimes(1);
    expect(errorNotification).toBeCalledWith({response: {status: 500}});
    expect(successNotification).toBeCalledTimes(0);
    expect(dom.find('button')).toHaveLength(2);
    expect(dom.find('button').at(0).text()).toBe('Save');
  });
  test('reset form on cancellation', async () => {
    await act(async () => {
      await dom.find('button').simulate('click');
      await dom.find('select[name="country"]').simulate('change', {target: {value: 'countryId1', name: 'country'}});
      await dom.find('input[name="phone"]').simulate('change', {target: {value: 7629, name: 'phone'}});
      await dom.find('select[name="city"]').simulate('change', {target: {value: 'cityId1', name: 'city'}});
      await dom.find('input[name="address1"]').simulate('change', {target: {value: 'port', name: 'address1'}});
    });
    dom.update();
    expect(dom.find('select[name="country"]').prop('value')).toBe('countryId1');
    expect(dom.find('input[name="phone"]').prop('value')).toBe(7629);
    expect(dom.find('select[name="city"]').prop('value')).toBe('cityId1');
    expect(dom.find('input[name="address1"]').prop('value')).toBe('port');
    await act(async () => {
      dom.find('button').at(1).simulate('click');
    });
    dom.update();
    expect(dom.find('select[name="phoneCode"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="phone"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="country"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="city"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="address1"]').prop('disabled')).toBeTruthy();
    expect(dom.find('input[name="address2"]').prop('disabled')).toBeTruthy();
    expect(dom.find('select[name="country"]').prop('value')).toBe('countryId1');
    expect(dom.find('input[name="phone"]').prop('value')).toBe(9352);
    expect(dom.find('select[name="city"]').prop('value')).toBe('cityId1');
    expect(dom.find('input[name="address1"]').prop('value')).toBe('edde');
    expect(dom.find('button')).toHaveLength(1);
    expect(dom.find('button').text()).toBe('Edit');
  });
});
