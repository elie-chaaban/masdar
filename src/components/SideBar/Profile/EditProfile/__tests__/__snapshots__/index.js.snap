// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Profile Component should match snapshot 1`] = `
<Memo(EditProfile)
  countries={
    Array [
      Object {
        "cities": Array [
          Object {
            "id": "cityId1",
            "name": "Beirut",
          },
        ],
        "code": "961",
        "id": "countryId1",
        "name": "Lebanon",
      },
      Object {
        "cities": Array [
          Object {
            "id": "cityId2",
            "name": "Dubai",
          },
        ],
        "code": "971",
        "id": "countryId2",
        "name": "United Arab Emirates",
      },
    ]
  }
  profile={
    Object {
      "address1": "edde",
      "address2": "",
      "city": Object {
        "country": Object {
          "cities": Array [
            Object {
              "id": "cityId1",
              "name": "Beirut",
            },
          ],
          "code": "961",
          "id": "countryId1",
        },
        "id": "cityId1",
        "name": "Dubai",
      },
      "email": "adel@gmail.com",
      "firstName": "adel96",
      "gender": "male",
      "mobileNumber": 9352,
      "mobileNumberCountryCode": "961",
    }
  }
>
  <Formik
    initialValues={
      Object {
        "address1": "edde",
        "address2": "",
        "city": "cityId1",
        "country": "countryId1",
        "email": "adel@gmail.com",
        "phone": 9352,
        "phoneCode": "961",
        "userName": "adel96",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "address2",
          "address1",
          "city",
          "country",
          "phoneCode",
          "phone",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "address1": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "address2": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_default": "",
            "_deps": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "city": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "country": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "phone": NumberSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "number",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "number",
          },
          "phoneCode": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Form
      className="content row"
    >
      <form
        action="#"
        className="content row"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <CustomInput
          enabled={false}
          name="userName"
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              userName
            </p>
            <Field
              className="form-control "
              disabled={true}
              name="userName"
              placeholder="userName"
              type="text"
            >
              <input
                className="form-control "
                disabled={true}
                name="userName"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="userName"
                type="text"
                value="adel96"
              />
            </Field>
          </div>
        </CustomInput>
        <CustomInput
          enabled={false}
          name="email"
          type="email"
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              email
            </p>
            <Field
              className="form-control "
              disabled={true}
              name="email"
              placeholder="email"
              type="email"
            >
              <input
                className="form-control "
                disabled={true}
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="email"
                type="email"
                value="adel@gmail.com"
              />
            </Field>
          </div>
        </CustomInput>
        <div
          className="mt-2 col-12"
        >
          <p>
            Phone
          </p>
        </div>
        <div
          className="col-6"
        >
          <div
            className="row"
          >
            <CountryCodeSelect
              enabled={false}
              name="phoneCode"
            >
              <div
                className=" col-3 pr-1"
              >
                <Field
                  name="phoneCode"
                >
                  <select
                    className="select-country "
                    disabled={true}
                    name="phoneCode"
                    onBlur={[Function]}
                    onChange={[Function]}
                    value="961"
                  >
                    <option
                      value=""
                    >
                      code
                    </option>
                    <option
                      key="961"
                      value="961"
                    >
                      961
                    </option>
                    <option
                      key="971"
                      value="971"
                    >
                      971
                    </option>
                  </select>
                </Field>
              </div>
            </CountryCodeSelect>
            <div
              className="col-9 pl-1"
            >
              <Field
                className="form-control "
                disabled={true}
                name="phone"
                placeholder="phone number"
                type="number"
              >
                <input
                  className="form-control "
                  disabled={true}
                  name="phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="phone number"
                  type="number"
                  value={9352}
                />
              </Field>
            </div>
            <div
              className="col-12"
            />
          </div>
        </div>
        <div
          className="col-6"
        />
        <CountrySelect
          countries={
            Array [
              Object {
                "cities": Array [
                  Object {
                    "id": "cityId1",
                    "name": "Beirut",
                  },
                ],
                "code": "961",
                "id": "countryId1",
                "name": "Lebanon",
              },
              Object {
                "cities": Array [
                  Object {
                    "id": "cityId2",
                    "name": "Dubai",
                  },
                ],
                "code": "971",
                "id": "countryId2",
                "name": "United Arab Emirates",
              },
            ]
          }
          enabled={false}
          name="country"
          onChange={[Function]}
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              Country
            </p>
            <Field
              name="country"
            >
              <select
                className="select-country "
                disabled={true}
                name="country"
                onBlur={[Function]}
                onChange={[Function]}
                value="countryId1"
              >
                <option
                  value=""
                >
                  select country
                </option>
                <option
                  key="countryId1"
                  value="countryId1"
                >
                  Lebanon
                </option>
                <option
                  key="countryId2"
                  value="countryId2"
                >
                  United Arab Emirates
                </option>
              </select>
            </Field>
          </div>
        </CountrySelect>
        <CitySelect
          cities={Array []}
          enabled={false}
          name="city"
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              City
            </p>
            <Field
              name="city"
            >
              <select
                className="select-city "
                disabled={true}
                name="city"
                onBlur={[Function]}
                onChange={[Function]}
                value="cityId1"
              >
                <option
                  value=""
                >
                  select city
                </option>
              </select>
            </Field>
          </div>
        </CitySelect>
        <CustomInput
          enabled={false}
          name="address1"
          title="address 1"
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              address 1
            </p>
            <Field
              className="form-control "
              disabled={true}
              name="address1"
              placeholder="address 1"
              type="text"
            >
              <input
                className="form-control "
                disabled={true}
                name="address1"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="address 1"
                type="text"
                value="edde"
              />
            </Field>
          </div>
        </CustomInput>
        <CustomInput
          enabled={false}
          name="address2"
          title="address 2"
        >
          <div
            className="mt-2 col-6"
          >
            <p>
              address 2
            </p>
            <Field
              className="form-control "
              disabled={true}
              name="address2"
              placeholder="address 2"
              type="text"
            >
              <input
                className="form-control "
                disabled={true}
                name="address2"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="address 2"
                type="text"
                value=""
              />
            </Field>
          </div>
        </CustomInput>
        <div
          className="col-12 mt-3"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </div>
      </form>
    </Form>
  </Formik>
</Memo(EditProfile)>
`;
